### DISCUSSION TITLE
Welcome to pyEfis Discussions!

### DISCUSSION THREAD

**Discussion by makerplane-jnicol** [makerplane/pyEfis#182] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by juanquy** [comment]
Hi, and I hope I'm writing in the right place.
Please look at the attached picture before, what you see is a Raspberry Pi 4 running the latest RasbianOS, pyEFIS, and FIXGateway from Snap Store, the newest application version as well.
I also have a Stratux box, both systems work very well together, I need help removing the right panel and buttons, I don't need it because I have another 5-inch screen for that matter, on this PI we only need the Horizont gauge.
As you see the background colors are not right, I don't know why and how it happens, I need help fixing it as well.

I hope to find help here.

peace and love
best
![Navi1](https://github.com/user-attachments/assets/3caa4155-cea2-4905-b9f7-0465c2de8cd1)

**Reply by e100** [reply]
I've been meaning to update the documentation to make it more clear how to customize things but I've not had the time.

First, most of what you need to know is here, this explains how to configure screens:
https://github.com/makerplane/pyEfis/blob/master/docs/screenbuilder.md

I highly suggest not making changes to any of the default files that are located in `~/makerplane/pyefis/config/` except for preferences.yaml.custom. If you need to change other files, make a copy and change the copy.

If you edit them, they will not get updated with future releases and those updates could include a very simple way to do exactly what you want and it will be easier for you to benefit from it if it is auto updated.

If you want to customize anything take a look at the preferences.yaml file.
Anything you want to change, copy it, then paste it into a new file named preferences.yaml.custom.
To remove the buttons you will need to edit each screen file.

The first screen is the file `screens/pfd.yaml`
Make a copy of that file and name it something else like `screens/my_pfd.yaml`
Now edit `preferences.yaml.custom` and add this text:
```
includes:
  SCREEN_PFD: screens/my_pfd.yaml
```

Now go customize the `screens/my_pfd.yaml` file and adjust to your liking.
To remove the buttons around line 23 you will find:
```
    - type: include,BUTTON_GROUP1
      row: 0
      column: 185.5
```

Add one new line `disabled: true` so it looks like:
```
    - type: include,BUTTON_GROUP1
      disabled: true
      row: 0
      column: 185.5
```

That should remove the buttons, but it will not adjust the gauges to occupy the space they took up.

To make those gauges wider a few lines up edit the filename `includes/arcs/vertical/four_high_two_states_preferences_ARC1-8.yaml`
Maybe name it `includes/arcs/vertical/my_four_high_two_states_preferences_ARC1-8.yaml`

Save the changes.
Now make a copy of that file.
Edit the copy you made `includes/arcs/vertical/my_four_high_two_states_preferences_ARC1-8.yaml`

Just a few lines down you will see the line `columns: 30`
Change it from 30 to 45.
Then look farther down the file where you see it again, change that one from 30 to 45 too.
Save the changes.

Well I hope the above is all correct, I could have made a typo or forgot about something as I did not actually test it.

My goal has been to make it so if you are happy with the default layouts but just need some minor changes, you can do that by only editing preference.yaml.custom.  I just do not know what all people might want to change.
But I also do not limit what you can do, you can change anything you want, make entirely new layouts. If you think other people would like your layout, we can included it in the code too!

Making a preferences option to disable the menu is very easy so I will likely add that, I am fairly confident I can even make it auto change the width of the gauges when you disable the menu. I hate to promise and not deliver but if I do have time to spare I will try to get this added over the weekend.

Seems like maybe we also need some screens that are made for portrait mode......
Got any suggestions?
If we had layouts for that mode, for each screen, it would be trivial to add a preferences option:
```
enabled:
  PORTRAIT_MODE: true
```

**Comment by juanquy** [comment]
Thank you, you are doing a great job, I will work on this on the weekend
and let you know my results

regards,
Juan

**Comment by juanquy** [comment]
BTW:

enabled:
  PORTRAIT_MODE: true

that is a great addition to the preference file! , I'm using the
offial 7 inc touch screen , but I also have another display like this
:

https://www.amazon.com/dp/B0C497LQFL?psc=1&ref=ppx_yo2ov_dt_b_product_details

only for engine gauges

I like how you have it so far, to me was hard to find how to trick the
interface, I think the documentation is a big step to enhance but as
an IT manager,I know how tedious and annoying is writing docu.

Also time consuming.

I may join you if I have free time.


best

**Comment by juanquy** [comment]
Thank you, Eric, see the attached image.
Following your guide, I removed the buttons, but nothing else.

ending I did this:


# Example primary flight display
PFD:
 include:
 - screens/virtualvfr_db.yaml
#  - HMI_ENCODER_BUTTONS
   #refresh_period: 0.25
 module: pyefis.screens.screenbuilder
 title: Single VFR 7inc display
 layout:
  # draw_grid: true
   rows: 110
   columns: 155
   display_state:
     interval: 3000
     states: 2
 instruments:
   - type: include,includes/ahrs/virtual_vfr.yaml
     row: 0
     column: 0
#      span:
#        rows: 0
#        columns: 100
#    - type:
include,includes/arcs/vertical/my_four_high_two_states_preferences_ARC1-8.yaml
#      row: 0
#      column: 155
#    - type: include,BUTTON_GROUP1
#      disabled: true
#      row: 0
#      column: 185.5
#    - type: include,includes/bars/vertical/six_wide_preferences_BAR1-6.yaml
#      row: 70
#      column: 155
#      display_state: 1
#    - type: include,includes/mgl/v16/square-active-radio-display.yaml
#      row: 70
#      column: 156
#      span:
#        rows: 40
#        columns: 45
#      display_state: 2
#      replace:
#        radio_id: 1


best

**Comment by e100** [comment]
@juanquy 

Lets take this discussion over to: #247 
You will find some useful information there
